segmenters/nlp/SpacySentencizer/__init__.py

```diff
@@ -0,0 +1,96 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class SpacySentencizer(BaseSegmenter):
+    """
+    :class:`SpacySentencizer` split the text on the doc-level
+    into sentences on the chunk-level with spaCy as the backend.
+    The splitting technique is divided into two different approaches:
+    1) Default Segmenter: Utilizes dependency parsing to determine the rule for
+        splitting the text. For more info please refer to https://spacy.io/api/sentencizer.
+    2) Machine Learning-based Segmenter: Utilizes SentenceRecognizer model trained
+        on a dedicated data created for sentence segmentation. For more info please
+        refer to https://spacy.io/api/sentencerecognizer.
+
+    :param lang: pre-trained spaCy language pipeline
+    :param use_default_segmenter: boolean to determine which segmentation algorithm to use,
+        by default False.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SPACY_COMPONENTS = [
+        "tagger",
+        "parser",
+        "ner",
+        "senter",
+        "entity_linker",
+        "textcat",
+        "entity_ruler",
+        "merge_noun_chunks",
+        "merge_entities",
+        "merge_subtokens",
+        "lemmatizer",
+        "attribute_ruler",
+    ]
+
+    def __init__(self, lang: str, use_default_segmenter: bool = False, *args, **kwargs):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+        self.lang = lang
+        self.use_default_segmenter = use_default_segmenter
+
+    def post_init(self):
+        import spacy
+
+        try:
```

Comment:
I am not an expert in Spacy, could you elaborate a bit on what are all these components and pipes?

Reply:
spaCy works in pipeline, so when a text enter spaCy it will be run through all of those pipes (pipes = pipelines). I'm thinking to disable all the pipelines and just enable ones that we need to remove extra latency

Reply:
but in this case we only want the pipeline related to segmenting right? or segmenting involves more than one?

Reply:
That's why I disabled the pipelines in the beginning as they're all pre-packaged with the models